packages/panzer/adapters-ioss/src/Panzer_IOSSConnManager.cpp

```diff
@@ -40,21 +40,970 @@
+        }
+        iossNodeIdToCoordinateMap[GlobalOrdinal(iossNodeBlockNodeIds[node])] = iossNodeCoordinates;
+      }
+    }
+
+  Teuchos::RCP<std::vector<GlobalOrdinal>> blockConnectivity;
+  Kokkos::DynRankView<double,PHX::Device> vertices("vertices",numElements,cornerNodesPerElement,dim);
+
+  blockConnectivity = iossConnectivity_.find(blockId)->second;
+  const Ioss::ElementTopology * iossElementBlockTopology = iossElementBlockTopologies_.find(blockId)->second;
+  int iossNodesPerElement = iossElementBlockTopology->number_nodes();
+  for (int elmtIdxInBlock = 0; elmtIdxInBlock < numElements; ++elmtIdxInBlock) {
+    for (int nodeIdxInElmt = 0; nodeIdxInElmt < cornerNodesPerElement; ++nodeIdxInElmt) {
+      iossNodeCoordinates = iossNodeIdToCoordinateMap[(*(blockConnectivity))[iossNodesPerElement*elmtIdxInBlock + iossElementBlockTopology->element_connectivity()[nodeIdxInElmt]]];
+      for (LocalOrdinal k = 0; k < dim; ++k) {
+        vertices(elmtIdxInBlock,nodeIdxInElmt,k) = iossNodeCoordinates[k];
+      }
+    }
+  }
+
+  // setup output array
+  points = Kokkos::DynRankView<double,PHX::Device>("points",numElements,idsPerElement,dim);
+  coordProvider.getInterpolatoryCoordinates(vertices,points);
+}
+
+void IOSSConnManager::clearLocalElementMapping()
+{
+
+   elementBlocks_.clear();
+   elmtLidToGid_.clear();
+   elmtLidToConn_.clear();
+   connSize_.clear();
+   //elmtToAssociatedElmts_.clear();
+}
+
+void IOSSConnManager::buildLocalElementMapping() {
+
+  std::string name;
+  std::vector<int> indices;
+  std::size_t element_count = 0;
+
+  // Initialize
+  clearLocalElementMapping();
+  ownedElementCount_ = 0;
+  elmtLidToGid_.clear();
+
+  for (Ioss::ElementBlock * iossElementBlock : iossElementBlocks_) {
+    name = iossElementBlock->name();
+	iossElementBlock->get_field_data("ids", indices);
+
+	// Ioss::GroupingEntity::get_field_data() will resize indices
+	// up or down as needed.
+	ownedElementCount_ += indices.size();
+
+    elementBlocks_[name] = rcp(new std::vector<LocalOrdinal>);
+    for (GlobalOrdinal element : indices) {
+      elementBlocks_[name]->push_back(element_count);
+      elmtLidToGid_.push_back(element);
+      ++element_count;
+    }
+  }
+
+  TEUCHOS_TEST_FOR_EXCEPTION(ownedElementCount_ != element_count, std::logic_error,
+		                     "Error, ownedElementCount_ != element_count.");
+
+  // Allocate space for elmtLidToConn_ and initialize to zero.
+  elmtLidToConn_.clear();
+  elmtLidToConn_.resize(ownedElementCount_,0);
+
+  // Allocate space for connSize_ and initialize to zero.
+  connSize_.clear();
+  connSize_.resize(ownedElementCount_,0);
+}
+
+void IOSSConnManager::buildEdgeFaceCornerNodeMapping() {
 
-#ifndef PANZER_ORDINAL64_IS_INT
-//template class IOSSConnManager<panzer::Ordinal64>;
-template class IOSSConnManager<int64_t>; // IOSS currently does not have a
-                                         // get_field_data<long long int>() defined.
```

Comment:
Fun fact: `int64_t` could be either `long` or `long long`.  (It's a typedef and doesn't have a required type.)